Content.Goobstation.Shared/Chemistry/SharedEnergyReagentDispenser.cs

```diff
@@ -0,0 +1,156 @@
+
+namespace Content.Goobstation.Shared.Chemistry
+{
+    /// <summary>
+    /// This class holds constants that are shared between client and server.
+    /// </summary>
+    public sealed class SharedEnergyReagentDispenser
+    {
+        public const string OutputSlotName = "beakerSlot";
+    }
+
+    [Serializable, NetSerializable]
+    public sealed class EnergyReagentDispenserSetDispenseAmountMessage : BoundUserInterfaceMessage
+    {
+        public readonly EnergyReagentDispenserDispenseAmount EnergyReagentDispenserDispenseAmount;
+
+        public EnergyReagentDispenserSetDispenseAmountMessage(EnergyReagentDispenserDispenseAmount amount)
+        {
+            EnergyReagentDispenserDispenseAmount = amount;
+        }
+
+        /// <summary>
+        ///     Create a new instance from interpreting a String as an integer,
+        ///     throwing an exception if it is unable to parse.
+        /// </summary>
+        public EnergyReagentDispenserSetDispenseAmountMessage(String s)
+        {
+            switch (s)
+            {
+                case "1":
+                    EnergyReagentDispenserDispenseAmount = EnergyReagentDispenserDispenseAmount.U1;
+                    break;
+                case "5":
+                    EnergyReagentDispenserDispenseAmount = EnergyReagentDispenserDispenseAmount.U5;
+                    break;
+                case "10":
+                    EnergyReagentDispenserDispenseAmount = EnergyReagentDispenserDispenseAmount.U10;
+                    break;
+                case "15":
+                    EnergyReagentDispenserDispenseAmount = EnergyReagentDispenserDispenseAmount.U15;
+                    break;
+                case "20":
+                    EnergyReagentDispenserDispenseAmount = EnergyReagentDispenserDispenseAmount.U20;
+                    break;
+                case "25":
+                    EnergyReagentDispenserDispenseAmount = EnergyReagentDispenserDispenseAmount.U25;
+                    break;
+                case "30":
+                    EnergyReagentDispenserDispenseAmount = EnergyReagentDispenserDispenseAmount.U30;
+                    break;
+                case "50":
+                    EnergyReagentDispenserDispenseAmount = EnergyReagentDispenserDispenseAmount.U50;
+                    break;
+                case "100":
+                    EnergyReagentDispenserDispenseAmount = EnergyReagentDispenserDispenseAmount.U100;
+                    break;
+                default:
+                    throw new Exception($"Cannot convert the string `{s}` into a valid ReagentDispenser DispenseAmount");
+            }
+        }
+    }
+
+    [Serializable, NetSerializable]
+    public sealed class EnergyReagentDispenserDispenseReagentMessage : BoundUserInterfaceMessage
+    {
+        public readonly string ReagentId;
+
+        public EnergyReagentDispenserDispenseReagentMessage(string reagentId)
+        {
+            ReagentId = reagentId;
+        }
+    }
```

Comment:
```suggestion
    [Serializable, NetSerializable]
    public sealed class EnergyReagentDispenserDispenseReagentMessage(string reagentId) : BoundUserInterfaceMessage
    {
        public readonly string ReagentId = reagentId;
    }
```
I <3 primary constructors